app/main.py

```diff
@@ -1,3 +1,20 @@
-def cache(func):
-    # Write your code here
-    pass
+from typing import Callable, Any
+from functools import wraps
+
+
+def cache(func: Callable) -> Callable:
+    dict__ = dict()
+
+    @wraps(func)
+    def inner(*args) -> Any:
+        nonlocal dict__
+
+        if not dict__.get((*args, inner.__name__,)) is None:
+            print("Getting from cache")
+            return dict__.get((*args, inner.__name__,))
```

Comment:
As i said above it is better to use `args` instead of `(*args, inner.__name__,)` as the key in the dictionary. This will reduce memory usage and simplify the code.